### PR TITLE
ci: add integration test to lint_test_build

### PR DESCRIPTION
**Pull request by BlairCurrey**
## Changes proposed in this pull request

- adds integration tests to ci

## Context

fixes https://github.com/interledger/rafiki/issues/2473

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass
- [ ] Bruno collection updated

### CODE REVIEW

File: test/integration/package.json
Comment (BlairCurrey) [inline test/integration/package.json:8]: from `docker-compose` -> `docker compose`

This is the same for me locally, but in github actions docker-compose is v1 and finds our compose files invalid (because of the top-level name attribute).

File: .github/workflows/integration_test.yml
Comment (BlairCurrey) [inline .github/workflows/integration_test.yml:8]: same event trigger as lint_test_build

File: .github/workflows/lint_test_build.yml
Comment (BlairCurrey) [inline .github/workflows/lint_test_build.yml:108]: Originally this was a seperate workflow file but I figured I'd include it here since it's the same event trigger.

I considered making this depend on auth, backend, token-introspection but decided not to because I think it gives us a useful signal if integration tests pass and auth/backend do not. This could happen if the failures just dont overlap with the tests cases, or if the backend/auth tests are flakey. Additionally, not making it depend on our longest running jobs ensures that we aren't increasing the time it takes to run CI. The integration tests take ~2 minutes, so they should finish before backend.

File: .github/workflows/lint_test_build.yml
Comment (BlairCurrey) [inline .github/workflows/lint_test_build.yml:118]: We're eventually going to use sudo and hostile to do this in the `run-tests` script but I think this will end up being permanent for the reason outlined in the end of the body in this issue: https://github.com/interledger/rafiki/issues/2555

**Comment by BlairCurrey** [comment]
Presumably backend (and auth) gql APIs will soon require a hydra service which is not included in these tests. So the integration tests will fail at that point and we will need to add hydra to the docker environment and handle authentication in the tests.

We should decide which we want to merge first. Either we merge this first and the authentication changes will be blocked until the testenv and tests handle it, or merge the authentication changes and this will be blocked.

**Comment by BlairCurrey** [comment]
> Presumably backend (and auth) gql APIs will soon require a hydra service which is not included in these tests. So the integration tests will fail at that point and we will need to add hydra to the docker environment and handle authentication in the tests.
> 
> We should decide which we want to merge first. Either we merge this first and the authentication changes will be blocked until the testenv and tests handle it, or merge the authentication changes and this will be blocked. It seems like we can probably piggy-back off either of these tasks to authenticate with the tests: #2568 #2557
> 
> I think being a much larger change, maybe we merge the authentication first (one less thing to implement there). But I don't know how close it is.

We talked about this in sprint planning and concluded the tests should be runnable w/out spinning up hydra. should work with a NODE_ENV that is test/development and using a special header. So the tests will break with those changes but we can simply update the header in the test requests. Given the changes required are small and dont look imminent I dont think we need to hold of on merging this IMO.

File: .github/workflows/lint_test_build.yml
Comment (sabineschaller) [reply .github/workflows/lint_test_build.yml:118]: I'm not sure I'm following. Why do we need it twice? Because that way, the script itself doesn't prompt?

**Review by sabineschaller — APPROVED** [review]
Just one clarifying question but otherwise, LGTM

**Review by mkurapov — DISMISSED** [review]
Approving for now, just one comment

File: .github/workflows/lint_test_build.yml
Comment (BlairCurrey) [reply .github/workflows/lint_test_build.yml:118]: Good questions - short answer is that I wonder if hostile will work fine in CI after all, in which case we can get rid of this step. I will test that.

The thinking was hostile will not work in github actions due to sudo prompt (needs confirming). Thus the extra step. And that appending to `/etc/hosts` and otherwise manually performing a "append if not exist" was better managed by something like hostile, thus not using  `echo "127.0.0.1 host.docker.internal" | sudo tee -a /etc/hosts`. And to answer Max's question, that does actually prompt from the command line - that's what makes me think my original assumptions about github actions and sudo might just be bad.

Also questioning if need hostile at this point. We are already checking for the existence of some line before setting a new one, so we shouldn't be adding duplicates. Beyond that and being cross-platform, which I don't think matters because we're already assuming a unix-like environment (including WSL) with this shell script and elswhere, it basically just does this:
- erroring if you try to overwrite local or localhost or the cli args are missing https://github.com/feross/hostile/blob/master/bin/cmd.js#L62-L70
- not setting if it exists (we're already checking that) https://github.com/feross/hostile/blob/master/index.js#L93
- adding a newline such that it preserves any existing whitespace at the end of the file.  https://github.com/feross/hostile/blob/master/index.js#L94-L101

So I'm not really seeing a value-add over checking if a line already exists and if not adding it.

Comment (BlairCurrey) [reply .github/workflows/lint_test_build.yml:118]: I guess I forgot where I was... editing `/etc/hosts` via hostile in `run-tests` is not in this branch and adding host step is required regardless here.

I will test using hostile in CI on that other branch when this is merged.

**Review by mkurapov — APPROVED** [review]

**Pull request merged**
